File: QMPlusVuePage/src/api/customer.js
import service from '@/utils/request'


export const getCustomerList = (data) => {
    return service({
        url: "/customer/list",
        method: 'post',
        data: data
    })
}

export const addCustomer = (data) => {
    return service({
        url: "/customer/add",
        method: 'post',
        data: data
    })
}

export const updateCustomer = (data) => {
    return service({
        url: "/customer/update",
        method: 'patch',
        data: data
    })
}

export const delCustomer = (data) => {
    return service({
        url: `/customer/delete`,
        method: 'delete',
        data: data
    })
}

export const getCustomerAddressList = (data) => {
    return service({
        url: "/customer/address",
        method: 'post',
        data
    })
}

export const addCustomerAddress = (data) => {
    return service({
        url: "/customerAddress/add",
        method: 'post',
        data
    })
}

export const updateCustomerAddress = (data) => {
    return service({
        url: "/customerAddress/update",
        method: 'patch',
        data
    })
}

export const delCustomerAddress = (data) => {
    return service({
        url: `/customerAddress/delete`,
        method: 'delete',
        data: data
    })
}

export const getCustomerById = (data) => {
    return service({
        url: `/customer/getbyid`,
        method: 'post',
        data: data
    })
}

export const getAddressById = (data) => {
    return service({
        url: "/customer/getbyid",
        method: 'post',
        data: data
    })
}

File: QMPlusServer/router/customer.go
package router

import (
	"gin-vue-admin/controller/api"
	"gin-vue-admin/middleware"
	"github.com/gin-gonic/gin"
)

func InitMyCustomerRouter(Router *gin.RouterGroup) {
	// 管理员操作
	customerAddressRouter := Router.Group("/customerAddress").Use(middleware.JWTAuth()).Use(middleware.CasbinHandler())
	{
		customerAddressRouter.POST("/address", api.GetAddressList)
		customerAddressRouter.POST("/add", api.AddAddress)
		customerAddressRouter.PATCH("/update", api.UpdateAddress)
		customerAddressRouter.DELETE("/delete", api.DeleteAddress)
		customerAddressRouter.POST("/getbyuserid", api.GetAddressListByUserId)
		customerAddressRouter.POST("/defaultAddress", api.SetDefaultAddress)
		customerAddressRouter.POST("/getbyid", api.GetAddressById)
	}

	customerRouter := Router.Group("/customer").Use(middleware.JWTAuth()).Use(middleware.CasbinHandler())
	{
		customerRouter.POST("list", api.GetCustomerList)
		customerRouter.PATCH("/update", api.UpdateCustomer)
		customerRouter.DELETE("/delete", api.DeleteCustomer)
		customerRouter.POST("/getbyid", api.GetCustomerById)
		customerRouter.POST("/add", api.AddCustomer)
	}

	customerOrderRouter := Router.Group("/customerOrder").Use(middleware.JWTAuth()).Use(middleware.CasbinHandler())
	{
		customerOrderRouter.POST("/order", api.GetOrderList)
		customerOrderRouter.DELETE("/delOrder", api.DeleteOrder)
		customerOrderRouter.POST("/getbyid", api.GetOrderByOrderId)

	}
	// 客户操作
	customerBusinessRouter := Router.Group("/cus").Use(middleware.JWTAuth())
	{
		customerBusinessRouter.POST("/upload", api.UploadCusHeaderImg)
		customerBusinessRouter.POST("/cart", api.GetCartList)
		customerBusinessRouter.POST("/addcart", api.AddCart)
		customerBusinessRouter.POST("/reduceCart", api.ReduceCart)
		customerBusinessRouter.POST("/delcart", api.DelCart)
		customerBusinessRouter.POST("/getprice", api.GetCoffeeSpecValue)
		customerBusinessRouter.POST("/checkStatus", api.CheckStatus)
		customerBusinessRouter.POST("/getAddress", api.GetAddressListByUserId)
		customerBusinessRouter.POST("/addAddress", api.AddAddress)
		customerBusinessRouter.POST("/getOrder", api.GetOrderListByOrderType)
		customerBusinessRouter.POST("/addOrder", api.AddOrder)
		customerBusinessRouter.POST("/delOrder", api.DeleteOrder)
		customerBusinessRouter.POST("/orderDetail", api.GetOrderByOrderId)
		customerBusinessRouter.POST("/defaultAddress", api.SetDefaultAddress)
	}
}
